helper/schema/core_schema.go

```diff
@@ -172,6 +172,53 @@ func (r *Resource) CoreConfigSchema() *configschema.Block {
 		}
 	}
 
+	// insert configured timeout values into the schema
+	if r.Timeouts != nil {
+		timeouts := configschema.Block{
+			Attributes: map[string]*configschema.Attribute{},
+		}
+
+		if r.Timeouts.Create != nil {
+			timeouts.Attributes["create"] = &configschema.Attribute{
+				Type:     cty.String,
+				Optional: true,
+			}
+		}
+
+		if r.Timeouts.Read != nil {
+			timeouts.Attributes["read"] = &configschema.Attribute{
+				Type:     cty.String,
+				Optional: true,
+			}
+		}
+
+		if r.Timeouts.Update != nil {
+			timeouts.Attributes["update"] = &configschema.Attribute{
+				Type:     cty.String,
+				Optional: true,
+			}
+		}
+
+		if r.Timeouts.Delete != nil {
+			timeouts.Attributes["delete"] = &configschema.Attribute{
+				Type:     cty.String,
+				Optional: true,
+			}
+		}
+
+		if r.Timeouts.Default != nil {
+			timeouts.Attributes["default"] = &configschema.Attribute{
+				Type:     cty.String,
+				Optional: true,
+			}
+		}
+
+		block.BlockTypes["timeouts"] = &configschema.NestedBlock{
```

Comment:
Is there already an `InternalValidate` step somewhere that prohibits a resource schema from containing a normal attribute called `timeouts`? Not absolutely required right now of course, since any existing one like that would already be broken in _some_ sense, but given that this feature is not well-known I worry that provider developers might try it in future and get confused about the weird behavior that will inevitably result, _particularly_ if they define `timeouts` as something that ends up in `block.Attributes` and so it'd actually be in _both_ maps. :scream:

Reply:
Now that you mention it, I think I'll add a check that this isn't overriding a timeouts block or attribute. This would probably leave it closer to the original behavior.